lib/sda/overlay.go

```diff
@@ -231,26 +238,40 @@ func (o *Overlay) Close() {
 	}
 }
 
-// CreateProtocol returns a fresh Protocol Instance with an attached
-// TreeNodeInstance
-func (o *Overlay) CreateProtocol(t *Tree, name string) (ProtocolInstance, error) {
-	tni := o.NewTreeNodeInstanceFromProtoName(t, name)
-	pi, err := ProtocolInstantiate(ProtocolNameToID(name), tni)
+// CreateProtocolSDA returns a fresh Protocol Instance with an attached
+// TreeNodeInstance. This protocol won't be handled by the service, but
+// only by the SDA.
+func (o *Overlay) CreateProtocolSDA(t *Tree, name string) (ProtocolInstance, error) {
+	return o.CreateProtocolService(ServiceID(uuid.Nil), t, name)
+}
+
+// CreateProtocolService adds the service-id to the token so the protocol will
+// be picked up by the correct service and handled by its NewProtocol method.
+func (o *Overlay) CreateProtocolService(sid ServiceID, t *Tree, name string) (ProtocolInstance, error) {
+	tni := o.NewTreeNodeInstanceFromService(t, t.Root, ProtocolNameToID(name), sid)
+	pi, err := ProtocolInstantiate(tni.token.ProtoID, tni)
 	if err != nil {
 		return nil, err
 	}
-	o.RegisterProtocolInstance(pi)
+	if err = o.RegisterProtocolInstance(pi); err != nil {
+		return nil, err
+	}
 	go pi.Dispatch()
 	return pi, err
 }
 
 // StartProtocol will create and start a P.I.
 func (o *Overlay) StartProtocol(t *Tree, name string) (ProtocolInstance, error) {
-	pi, err := o.CreateProtocol(t, name)
+	pi, err := o.CreateProtocolSDA(t, name)
 	if err != nil {
 		return nil, err
 	}
-	go pi.Start()
+	go func() {
+		err := pi.Start()
+		if err != nil {
+			dbg.Error("Error while starting:", err)
```

Comment:
It depends in the situation, here we are forced to call Start() in a go-routine because the go routine can take time, but we don't want the main *thread* to hang up on it waiting on the error response (otherwise it's useless to use channels). we have this chiken-and-egg problem here, and it does not make sense for me to have the Start return an error because we can't use it, and we can't make a meaning out of it (it's not the sda code (network/tree etc), it's a protocol, so it should be a protocol that handle its own error, not sda)